source/common/http/http1/codec_impl.cc

```diff
@@ -376,16 +377,83 @@ void ServerConnectionImpl::onEncodeComplete() {
   }
 }
 
+// handlePath computes the correct :authority and :path headers based on 7230#5.7 and 7230#6
+// Throws CodecProtocolException on bad request
+void ServerConnectionImpl::handlePath(HeaderMapImplPtr& headers, bool is_connect) {
+  // TODO(mattwoodyard) - check configuration option if forward proxy is enabled
+  // In practice there will only ever be one scheme delivered for forward proxy
+  // support
+  HeaderString path(Headers::get().Path);
+
+  // The url is relative. Nothing to do here.
+  if (active_request_->request_url_.c_str()[0] == '/' ||
+      active_request_->request_url_.c_str()[0] == '*') {
+    headers->addViaMove(std::move(path), std::move(active_request_->request_url_));
+    return;
+  }
+
+  if (!codec_settings_.allow_connect_ && !codec_settings_.allow_absolute_url_) {
+    return;
```

Comment:
I'm curious about this return.  If we know this is an absolute URL and it's not going to be handled, wouldn't it be reasonable to sendProtocolError here?  It just seems odd to neither set up the path (as we always did in the prior code path) or send a response.  If this is the correct behavior maybe add a comment explaining why?